sw/host/tests/xmodem/xmodem.rs

```diff
@@ -101,38 +122,54 @@ impl XmodemFirmware {
 
 // The xmodem_{read,write} functions provide the interface to the low-level C implementation to
 // interact with the `Uart` device provided to the `XmodemFirmware` struct.
+// SAFETY: `data` must be valid for `len` bytes.
```

Comment:
The safety contract is usually documented like this:
```rust
/// # Safety
///
/// `data` must be valid for `len` bytes.
```

Reply:
Thanks, converted to a doc comment made the safety part a header.